Set images for all states when using wxAnyButton::SetBitmap()

MaartenBent opened the pull request:
Invalidate current bitmap data, so all states will be updated when calling SetBitmapLabel().

Fixes #22109

Note that this change is in a common header, I haven't checked osx or gtk toolkits yet for regressions.

vadz commented:
Thanks for fixing this!

I agree that `SetBitmap()` should change all bitmaps, but IMHO so should `SetBitmapLabel()` as I think these methods are supposed to do the same thing.

Also, at least in wxGTK removing the bitmap and then adding it is more expensive than just changing the bitmap and while this might not be noticeable, I think it would still be better to avoid doing it.

So I think it would be better to handle the normal bitmap specially in `DoSetBitmap()` implementation in wxMSW and reset all the other bitmaps when it's changed. What do you think?

MaartenBent commented:
Sure, having a solution only in wxMSW also has my preference. I'll update this PR.

MaartenBent committed:
Set images for all states when using wxAnyButton::SetBitmap()

Delete the wxButtonImageData when setting the normal bitmap,
so when the new one is created all states are initialized.

Fixes #22109

vadz commented on src/msw/anybutton.cpp:
```diff
@@ -812,29 +812,29 @@ wxBitmap wxAnyButton::DoGetBitmap(State which) const
     if ( !bitmapBundle.IsOk() )
     {
-        if ( m_imageData  )
+        if ( m_imageData && which != State_Normal )
```
This doesn't seem to be necessary? I.e. if `which == State_Normal` then `m_imageData` is already null anyhow.

I'll remove this change if you don't mind.

MaartenBent replied:
You're right. I just kept all the checks from the original if-statements. This one seems unnecessary now.

vadz reviewed (commented):
Thanks! I'm going to merge it unless you object to the proposed change below.
